internal/info/info.go

```diff
@@ -26,9 +26,25 @@ import (
 	"strings"
 	"sync"
 
+	"github.com/vdaas/vald/internal/errors"
 	"github.com/vdaas/vald/internal/log"
 )
 
+// InformationProvider represents the runtime information provider.
+type InformationProvider interface {
+	String() string
+	Get() Detail
+}
+
+type info struct {
+	detail   Detail
+	prepOnce sync.Once
+
+	// runtime functions
+	rtCaller    func(skip int) (pc uintptr, file string, line int, ok bool)
+	rtFuncForPC func(pc uintptr) *runtime.Func
```

Comment:
Should we use a function type field for mocking?
Did we talk about avoiding the use of function type fields to mock?
if I made a mistake, I am sorry.

Reply:
I asked PO and he think it is over implementation. 
ref: https://vdaas.slack.com/archives/CRGKSQPBP/p1606967701020600?thread_ts=1606874350.009200&cid=CRGKSQPBP